File: SocialNetwork.WebApp/src/types/message.ts
import { MediaType } from "../enums/media";
import { MessageType } from "../enums/message-type";
import { UserResource } from "./user";

export type MessageResource = {
    id: string;
    sender: UserResource;
    senderId: string;
    content: string;
    medias: MessageMediaResource[]
    sentAt: Date;
    chatRoomId: string;
    status: string;
    reads?: ReadStatusResource[];
    messageType?: MessageType

    isRemove?: boolean;
    memberId?: string;

    isFetch?: boolean;
}

export type ReadStatusResource = {
    user: UserResource;
    readAt: Date;
    userId: string;
}


export type MessageMediaResource = {
    id: string;
    mediaUrl: string;
    mediaType: MediaType
}

File: SocialNetwork.Domain/Entity/Message.cs


using SocialNetwork.Domain.Abstractions;

namespace SocialNetwork.Domain.Entity
{
    public class Message : BaseAuditableEntity
    {
        public string Content { get; set; } 
        public string MessageType { get; set; }
        public string? SenderId { get; set; } 
        public User Sender { get; set; }
        public Guid ChatRoomId { get; set; } 
        public ChatRoom ChatRoom { get; set; }
        public ICollection<MessageMedia> Medias { get; set; }
        public ICollection<MessageReadStatus> Reads { get; set; }
        public DateTimeOffset SentAt { get; set; }
    }
}
